stl/inc/memory

```diff
@@ -201,52 +322,156 @@ template <class _InIt, class _Diff, class _NoThrowFwdIt>
+            }
+
+            _Adl_verify_range(_First2, _Last2);
+            auto _IFirst      = _Get_unwrapped_n(_STD move(_First1), _Count);
+            auto _OFirst      = _Get_unwrapped(_STD move(_First2));
+            const auto _OLast = _Get_unwrapped(_STD move(_Last2));
+            if constexpr (_Ptr_move_cat<_It, _Out>::_Really_trivial) {
+                _OFirst = _Copy_memcpy_common(_IFirst, _IFirst + _Count, _OFirst, _OLast);
+            } else {
+                _Uninitialized_backout _Backout{_STD move(_OFirst)};
+
+                for (; _Count > 0 && _Backout._Last != _OLast; --_Count, (void) ++_IFirst) {
+                    _Backout._Emplace_back(_RANGES iter_move(_IFirst));
+                }
+
+                _OFirst = _Backout._Release();
+            }
+
+            _Seek_wrapped(_First1, _IFirst);
+            _Seek_wrapped(_First2, _OFirst);
+            return {_STD move(_First1), _STD move(_First2)};
+        }
+    };
+
+    inline constexpr _Uninitialized_move_n_fn uninitialized_move_n{_Not_quite_object::_Construct_tag{}};
+
+    // VARIABLE ranges::uninitialized_fill
+    class _Uninitialized_fill_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <_No_throw_forward_iterator _It, _No_throw_sentinel_for<_It> _Se, class _Ty>
+            requires constructible_from<iter_value_t<_It>, const _Ty&>
+        _It operator()(_It _First, _Se _Last, const _Ty& _Val) const {
+            // clang-format on
+            _Adl_verify_range(_First, _Last);
+            auto _UResult = _Uninitialized_fill_unchecked(
+                _Get_unwrapped(_STD move(_First)), _Get_unwrapped(_STD move(_Last)), _Val);
+
+            _Seek_wrapped(_First, _STD move(_UResult));
+            return _First;
+        }
+
+        // clang-format off
+        template <_No_throw_forward_range _Rng, class _Ty>
+            requires constructible_from<range_value_t<_Rng>, const _Ty&>
+        borrowed_iterator_t<_Rng> operator()(_Rng&& _Range, const _Ty& _Val) const {
+            // clang-format on
+            return _Rewrap_iterator(_Range, _Uninitialized_fill_unchecked(_Ubegin(_Range), _Uend(_Range), _Val));
+        }
+
+    private:
+        template <class _It, class _Se, class _Ty>
+        _NODISCARD static _It _Uninitialized_fill_unchecked(_It _OFirst, const _Se _OLast, const _Ty& _Val) {
+            _STL_INTERNAL_STATIC_ASSERT(_No_throw_forward_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(_No_throw_sentinel_for<_Se, _It>);
+            _STL_INTERNAL_STATIC_ASSERT(constructible_from<iter_value_t<_It>, const _Ty&>);
+
+            if constexpr (_Fill_memset_is_safe<_It, _Ty>) {
+                const auto _OFinal = _RANGES next(_OFirst, _STD move(_OLast));
+                const auto _Diff   = static_cast<size_t>(_OFinal - _OFirst);
+                _CSTD memset(_OFirst, static_cast<unsigned char>(_Val), _Diff);

```

Comment:
```suggestion
                _Fill_memset(_OFirst, _Val, _Diff);
```

Reply:
Excellent catch - thank you for finding what we missed! :heart_eyes_cat: